### ISSUE TITLE
Add EOL option

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#2957]
twitter thread https://twitter.com/domenic/status/697921719529398272
![image](https://cloud.githubusercontent.com/assets/172399/13174587/02e38508-d705-11e5-92c5-372e75ff500d.png)

**Comment by stkb** [comment]
+1 An option to change the default on Windows would be nice.

**Comment by egamma** [comment]
![image](https://cloud.githubusercontent.com/assets/172399/13174577/eddcb6ac-d704-11e5-91fe-27abb04a3822.png)

**Comment by glen-84** [comment]
Duplicate of #1546?

**Comment by alexandrudima** [comment]
639a3cbcb3ac118dfaba7443e2ee3b502ff34d21

**Comment by egamma** [comment]
@alexandrudima i'll add this to the feb plan, so that we do not forget to release note the feature

**Comment by edgr-sanchez** [comment]
Is there a way to set LF line endings as default? 

I looked in Preferences > User Settings but didn't see a setting for this.

My Surface + Visual Studio Code are used as extensions of my Linux/Mac environment. I'll never need CRLF so I have to constantly change from CRLF to LF when I create new documents. If I forget to do this it causes trouble.

**Comment by glen-84** [comment]
@edgr-sanchez

```json
"files.eol": "\n"
```

**Comment by edgr-sanchez** [comment]
@glen-84 

Perfect, thank you!

**Comment by NoobJohn** [comment]
@glen-84 
nice job! thanks

**Comment by 624508914** [comment]
thank you @glen-84

**Comment by tharrison159** [comment]
I have "files.eol": "\n" set in my user preferences but whenever I open a project it always defaults to clrf, any ideas why it's still doing this?

**Comment by 624508914** [comment]
@tharrison159 yes，a project in vscode it always defaults to crlf, because vscode set user preferences is  into a project

**Comment by tharrison159** [comment]
I am working on a project that is shared between unix systems and windows so the line breaks need to be LF not CLRF, is there anyway to convert the whole project to LF?

**Comment by 624508914** [comment]
use git and set `git config --global core.autocrlf true / false / input`

**Comment by 624508914** [comment]
@tharrison159 What are your FB account or TW account

**Comment by jsilvermist** [comment]
This should be better documented, nothing comes up when searching for `lf` or `crlf` in the settings.

**Comment by alexandrudima** [comment]
Pushed a change to improve the documentation

<img width="323" alt="screen shot 2017-05-20 at 11 45 13" src="https://cloud.githubusercontent.com/assets/5047891/26274770/d6e3959c-3d51-11e7-9c00-735a9f37f900.png">

**Comment by strattonbrazil** [comment]
Something that's not entirely obvious to many, but it seems that if there are already CRLFs in the existing file, VSC will fallback to using those.  After seeing CRLFs introduced in the change log, I tried a different file and didn't see them added.  I also saw in settings it was already set to use LF.  When I ran dos2unix on the offending file and then when back and added a new line, I no longer saw the CRLFs.  This seems to match the functionality I've seen on some other editors.  

TL;DR If you're seeing CRLFs and your settings are using LF, you might need to remove the existing CRLFs in the file or else the editor will fall back to using those for consistency across lines.

**Comment by grobmeier** [comment]
Actually @strattonbrazil is right. I had the same problem: there was a CRLF in a css file which caused VSC to write CRLF, even when I had `"files.eol": "\n",`

Running "dos2unix" fixed the problem entirely. The git config didn't help, it just fixes the changed lines (I think)

**Comment by rsmolkin** [comment]
So I'm having the same problem.  Working on a team where some have Macs and some have Windows.  My files.eof is set to "\n" but every time I open a file it still shows crlf in the bottom blue bar and my eslint freaks out, so I have to change it every time I open a file.  How can we set this to be default not just on new files but for every file in a project?

**Comment by drkmullins** [comment]
Was having similar issues to @rsmolkin and seemed to resolve it by adding ```*.js       text eol=lf```
to my .gitattributes file as mentioned in the [linebreak-style rule docs](https://eslint.org/docs/rules/linebreak-style)

**Comment by rsmolkin** [comment]
@drkmullins Thanks for the find.  Can you clarify if this is for the global Git directory where Git is installed, or local to each project in the .git/info/ directory.  The only file I see in there is exclude, but I guess I can add .gitattributes with that one line in it?

**Comment by bialix** [comment]
Is it possible to configure line-endings settings per file type? E.g. LF for python files, but CRLF for batch files?
